gpytorch/lazy/mul_lazy_variable.py

```diff
@@ -18,18 +18,30 @@ def __init__(self, *lazy_vars):
         """
         lazy_vars = list(lazy_vars)
         if len(lazy_vars) == 1:
-            raise RuntimeError("MulLazyVariable should have more than one lazy variables")
-
-        for i, lazy_var in enumerate(lazy_vars):
-            if not isinstance(lazy_var, LazyVariable):
-                if isinstance(lazy_var, Variable):
-                    lazy_vars[i] = NonLazyVariable(lazy_var)
-                else:
-                    raise RuntimeError("All arguments of a MulLazyVariable should be lazy " "variables or variables")
+            if isinstance(lazy_vars[0], NonLazyVariable):
+                self._non_lazy_self = lazy_vars
+            else:
+                raise RuntimeError("MulLazyVariable should have more than one lazy variables")
+        else:
+            for i, lazy_var in enumerate(lazy_vars):
+                if not isinstance(lazy_var, LazyVariable):
+                    if isinstance(lazy_var, Variable):
+                        lazy_vars[i] = NonLazyVariable(lazy_var)
+                    else:
+                        raise RuntimeError("All arguments of a MulLazyVariable should be lazy variables or variables")
 
         super(MulLazyVariable, self).__init__(*lazy_vars)
         self.lazy_vars = lazy_vars
 
+    @property
+    def non_lazy_self(self):
+        if hasattr(self, "_non_lazy_self"):
+            return self._non_lazy_self[0]
```

Comment:
Will `_non_lazy_self` ever contain more than one element? If not, why make it a list?

Reply:
Because setting an argument in `MulLazyVariable` to a LazyVariable directly would lead to infinite recursion due to how `LazyVariable.__setattr__` works 👎 🤒 .

...I'll raise an issue to rewrite `MulLazyVariable` to be less gross, haha.

Reply:
@gpleiss

Reply:
make sure to create a degrossify label for it :)